### PR TITLE
Fix navigation after enabling plugin

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #12886 

Seems like there's a Qt bug / strange behaviour, which was just revealed by #12854. Problem is that the list view items have not yet been loaded at the time that we call `requestActive`, even though the items _should_ exist. Apparently, assigning something to the `currentIndex` property of the grid view somehow forces the items to be loaded. 

Also tweaks the navigation/selection borders of Plugins and Scores thumbnails yet a bit further.

### CODE REVIEW

**Comment by DmitryArefiev** [comment]
Hi @cbjeukendrup, it this ready for testing?

**Comment by DmitryArefiev** [comment]
OK, got it! I'll test after review from @Eism

File: src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml [hunk 1]
```diff
@@ -39,6 +39,6 @@ Rectangle {
     border.color: ui.theme.fontPrimaryColor
     border.width: ui.theme.navCtrlBorderWidth
     radius: Number(parent.radius) > 0
-            ? (drawOutsideParent ? parent.radius + root.border.width : parent.radius)
```
Comment (Eism) [inline src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml:42]: Please explain why this change is needed?

File: src/plugins/qml/MuseScore/Plugins/PluginsPage.qml [hunk 2]
```diff
@@ -102,9 +102,7 @@ Item {
 
         anchors.top: parent.top
         anchors.left: parent.left
-        anchors.leftMargin: root.sideMargin
```
Comment (Eism) [inline src/plugins/qml/MuseScore/Plugins/PluginsPage.qml:105]: It is better to return as it was. Like in scores and learn pages

**Review by Eism — APPROVED** [review]

**Review by Eism — CHANGES_REQUESTED** [review]
Sorry, approved accidentally

File: src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml (hunk 1 above)
Comment (cbjeukendrup) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml:42]: It fixes the corner radius of the navigation border for the thumbnails on the Scores page. In ScoreItem, we set the margin for the NavigationBorder explicitly, to create some padding between the border and the thumbnail, and with this change, the corner radius of the border is adjusted accordingly.

File: src/plugins/qml/MuseScore/Plugins/PluginsPage.qml (hunk 2 above)
Comment (cbjeukendrup) [reply src/plugins/qml/MuseScore/Plugins/PluginsPage.qml:105]: The change is necessary because otherwise the navigation/selection border would be cut off. We can't set `clip` to `false`, because that will result in the items being visible above and below the scrollable area.

(In the current master branch, the clipping problem does not occur, because the items have an inset of 4px. But that way, they are not aligned with the "Enabled"/"Disabled" title text, which looks a bit ugly. )

File: src/plugins/qml/MuseScore/Plugins/internal/PluginsListView.qml [hunk 3]
```diff
@@ -51,6 +51,10 @@ Column {
     signal navigationActivated(var itemRect)
 
     function focusOnFirst() {
+        // Strange hack to force the view to load the items. Without this, `view.itemAtIndex(0)` might be null
```
[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/plugins/qml/MuseScore/Plugins/internal/PluginsListView.qml:54]: I'm afraid no, but I'll take a look at the Qt source code, maybe that gives some insight why this hack works and if there's a better way

[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/plugins/qml/MuseScore/Plugins/internal/PluginsListView.qml]: Seems I found something that works

File: src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml (hunk 1 above)
Comment (Eism) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml:42]: What's wrong with the borders for the Score item in master? Looks nice

image

File: src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml [hunk 4]
```diff
@@ -50,29 +50,17 @@ Item {
...
         width: parent.width
         height: 144
 
-        color: "transparent"
-        radius: root.radius + border.width
-
-        border.color: ui.theme.fontPrimaryColor
-        border.width: root.selected ? 2 : 0
-
-        NavigationFocusBorder {
```
Comment (Eism) [inline src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml:66]: Why you decided to use a custom navigation border instead of this component?

Comment (cbjeukendrup) [reply src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml:66]: The border for navigation and selection looks exactly the same, so I thought it is redundant to use two separate rectangles for these purposes. If you prefer, I could reuse `NavigationFocusBorder` and just override its `visible` property to make it also visible when the item is selected.

File: src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml (hunk 1 above)
Comment (cbjeukendrup) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml:42]: It's subtle, _very_ subtle, but it is not entirely correct. I've tried to indicate the difference in this image:
Scherm­afbeelding 2022-08-29 om 10 57 24
With the old version (left), it will look like the padding between the border and the thumbnail is a bit bigger in the corners. With the new version, it will look and feel like 2px all around.

File: src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml (hunk 4 above)
Comment (Eism) [reply src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml:66]: They perform different functions and it is better not to mix them. 
It should be clear from the code where the selection rectangle is and where the navigation is.

File: src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml (hunk 1 above)
Comment (Eism) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml:42]: ok, thanks for clarification

Comment (cbjeukendrup) [reply src/framework/uicomponents/qml/MuseScore/UiComponents/NavigationFocusBorder.qml:42]: I decided to add a dedicated "padding" property to `NavigationFocusBorder`, to make it more clear what we're doing.

File: src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml (hunk 4 above)
Comment (cbjeukendrup) [reply src/plugins/qml/MuseScore/Plugins/internal/PluginItem.qml:66]: Okay, I separated them!

**Review by Eism — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
Tested #12886 on Win10, Mac12 - **FIXED**  
(can't check linux build, something wrong with recent CI builds for linux, will check on master)

**Comment by cbjeukendrup** [comment]
> something wrong with recent CI builds for linux

Oh no... I'm afraid this is related to #12963...
